src/goto-symex/renamed.cpp

```diff
@@ -14,9 +14,6 @@ Author: Romain Brenguier, romain.brenguier@diffblue.com
 #include <util/std_expr.h>
 bool is_l1_renamed(const exprt &expr)
 {
-  if(!is_l2_renamed(expr.type()))
-    return false;
```

Comment:
The claim made in the commit message seems wrong: the type may well be renamed to L2 even when the object itself is only L1. On the contrary, it actually makes no sense for the type to be renamed to L1 ever.

Reply:
Ok I didn't know that the type could be L2 while the object L1. But is it true that L1 object always have L2 types? It sounds wrong to me. It seems that when an `ssa_exprt` is renamed to L1, its type is also renamed to L1.

Reply:
Renaming to L1 effectively is instantiation/determines which instance of a named object we are using. At that point the type must be complete, i.e., must be L2.

Reply:
It seems to be the case in symex_decl but not in symex_dereference and the other places where L1 renaming is used. Should we make `rename<L1>` apply `L2` renaming on the type to ensure this is consistent?